service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderControllerTest.java

```diff
@@ -216,4 +204,104 @@ private CallbackRequest buildCallbackRequest() {
             .put("localAuthorityOrCafcass", LOCAL_AUTHORITY_NAME)
             .build();
     }
+
+    @Nested
+    class MidEvent {
+        private final byte[] pdf = {1, 2, 3, 4, 5};
+        private Document document;
+
+        @BeforeEach
+        void setUp() {
+            document = document();
+            DocmosisDocument docmosisDocument = new DocmosisDocument("order.pdf", pdf);
+
+            given(docmosisDocumentGeneratorService.generateDocmosisDocument(any(), any())).willReturn(docmosisDocument);
+            given(uploadDocumentService.uploadPDF(any(), any(), any(), any())).willReturn(document);
+        }
+
+        @Test
+        void midEventShouldGenerateOrderDocumentWhenOrderTypeIsBlankOrder() {
+            AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(generateBlankOrderCaseDetails());
+
+            CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+            verify(uploadDocumentService, times(1)).uploadPDF(userId, userAuthToken, pdf, "blank_order_c21.pdf");
+
+            assertThat(caseData.getOrderTypeAndDocument().getDocument()).isEqualTo(expectedDocument());
+        }
+
+        @Test
+        void midEventShouldGenerateOrderDocumentWhenOrderTypeIsCareOrderWithFurtherDirections() {
+            final AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(
+                generateCareOrderCaseDetails(true));
+
+            verify(docmosisDocumentGeneratorService, times(1)).generateDocmosisDocument(any(), any());
+            verify(uploadDocumentService, times(1)).uploadPDF(userId, userAuthToken, pdf, "care_order.pdf");
+
+            final CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+            assertThat(caseData.getOrderTypeAndDocument().getDocument()).isEqualTo(expectedDocument());
+        }
+
+        @Test
+        void midEventShouldNotGenerateOrderDocumentWhenOrderTypeIsCareOrderWithNoFurtherDirections() {
+            postMidEvent(generateCareOrderCaseDetails(false));
+
+            verify(docmosisDocumentGeneratorService, never()).generateDocmosisDocument(any(), any());
+            verify(uploadDocumentService, never()).uploadPDF(any(), any(), any(), any());
+        }
+
+        @AfterEach
+        void tearDown() {
+            reset(docmosisDocumentGeneratorService);
+            reset(uploadDocumentService);
+        }
```

Comment:
why do we need to tear down?

Reply:
A bug with Spring means that mock beans aren't reset after each run, as such this invocation count doesn't reset and the tests fail, this does what is supposed to happen but manually

Reply:
https://github.com/spring-projects/spring-boot/issues/12470
The provided solutions here didn't work for me either

Reply:
Renamed the method to be a bit more clear, tearDown was just the auto generated name